### PR TITLE
Assignment 1: Gales & Miller

### PR DESCRIPTION
**Pull request by webnoben**

### CODE REVIEW

**Comment by tienv9** [comment]
Looking good overall, and the added color makes it much clearer to see if the answer is correct or incorrect. One thing to note: please base your PR on the correct branch instead of main, as it makes it difficult to track the changes. If you need to create a new branch, follow these steps: click on Branches > View All Branches > New Branch, then select the class repository as the source and name your branch.

Assignment Details
Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Come up with a new feature for the application
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ❌

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if (userGuess == question.CorrectAnswerIndex)
+        {
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine("Correct");
+            Console.ForegroundColor = ConsoleColor.White;
+            return true;
+        }
+        Console.ForegroundColor= ConsoleColor.Red;
+        Console.WriteLine("Incorrect");
+        Console.ForegroundColor = ConsoleColor.White;
```
Comment (tienv9) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:54]: Great addition to the code to improve visibility for the result.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public static void Main(string[] args)
 
     public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
     {
-        return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+        return ((double)numberCorrectAnswers / (double)numberOfQuestions * 100) + "%";
```
Comment (BenKnoll3) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:27]: Consider a space after "(double)" to help code readability.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 3]
```diff
@@ -44,11 +44,14 @@ public static bool DisplayResult(string userGuess, Question question)
     {
         if (userGuess == question.CorrectAnswerIndex)
         {
+            Console.ForegroundColor = ConsoleColor.Green;
```
Comment (BenKnoll3) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:47]: I like this feature. Simple, yet adds stronger feedback for the user.

**Review by BenKnoll3 — COMMENTED** [review]
Your changes look solid, however it was a bit hard to read the git diff because the PR is comparing to "Intellitect-Samples:main" instead of "Intellitect-Samples:Assignment1". This is the only major change you need to make, other than that good job!

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 4]
```diff
@@ -21,7 +21,10 @@ public static void Main(string[] args)
 
     public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
     {
-        return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+        if (numberOfQuestions == 0)
+            throw new DivideByZeroException("numberOfQuestions cannot be 0");
```
Comment (BenjaminMichaelis) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:25]: Isn't this thrown already if this is attempted? What value does this add?
```suggestion
```

**Pull request closed without merging**